source/agora/utils/PrettyPrinter.d

```diff
@@ -64,6 +64,8 @@ public auto prettify (T) (const auto ref T input) nothrow
         return TransactionFmt(input);
     else static if (is(T : const Block))
         return BlockFmt(input);
+    else static if (is(T : const (Block)[]))
+        return BlocksFmt(input);
```

Comment:
If you change it to:

```D
else static if (isInputRange!T && is(ElementType!T : Block))
    return BlocksFmt!T(input)
```

then it will accept any range type, and not just arrays.

Reply:
Yes I had this in mind for next time I wanted an array printed - or should it be third time with the rule of three applied? ;-)

Reply:
My mistake, this is only for blocks still. But I think we can make it templated like you mentioned later.

Reply:
Updated to handle any range.